src/firefox/index.js

```diff
@@ -239,8 +244,34 @@ export type UseProfileParams = {
   app?: PreferencesAppName,
   configureThisProfile?: ConfigureProfileFn,
   customPrefs?: FirefoxPreferences,
+  createProfileFinder?: typeof defaultCreateProfileFinder,
 };
 
+export function defaultCreateProfileFinder() {
+  const finder = new FirefoxProfile.Finder();
+  const readProfiles = promisify(finder.readProfiles, finder);
+  return {
+    getPath: promisify(finder.getPath, finder),
+    hasProfileName: async (profileName: string) => {
+      try {
+        await fs.stat(path.join(FirefoxProfile.Finder.locateUserDirectory(),
+                                                              'profile.ini'));
+      } catch (error) {
+        if (isErrorWithCode('ENOENT', error)) {
+          log.info('No firefox profiles exist');
```

Comment:
Here, we have already checked that "profiles.ini" doesn't exist (which should probably be a warning instead of an info message), but we are going to execute `await readProfiles()` at line 266, which will raises an exception because of that.

And so I think that it would be reasonable to `return false` after logging a warning (no profile name can be found if no `"profiles.ini"` has been found).